agentcfg/model.go

```diff
@@ -34,43 +35,34 @@ type Settings map[string]string
 
 // NewDoc unmarshals given byte slice into t Doc instance
 func NewDoc(in []byte) (*Doc, error) {
-	var doc Doc
 	if len(in) == 0 {
-		return &doc, nil
+		return &Doc{}, nil
 	}
 	var tmp tmpDoc
 	if err := json.Unmarshal(in, &tmp); err != nil {
 		return nil, err
 	}
-	if err := json.Unmarshal(tmp.Source.Settings, &doc.Settings); err != nil {
-		return nil, err
+
+	h := md5.New()
+	var keys []string
+	for k := range tmp.Source.Settings {
+		keys = append(keys, k)
 	}
-	doc.ID = tmp.hash()
-	return &doc, nil
+	sort.Strings(keys)
+	out := make(map[string]string)
+	for _, k := range keys {
+		out[k] = fmt.Sprintf("%v", tmp.Source.Settings[k])
```

Comment:
any concerns about nested keys here?

Reply:
Thanks for raising the question again. I checked back with the agent devs, and they expect nested keys to be returned in the format `"a.b.c": "val"` rather then `"a": "{b: {c: val}}"`.
I now updated the handling accordingly, for a config of form:
```
"settings" : {
  "sample_rate" : 0.7,
  "nested_1" : {
    "k1" : [1,2,3],
    "k2" : "test",
    "nested_2" : {
      "k3" : 56
    } 
  }
}
```
we  now return:
```
{
  "nested_1.k1": "1,2,3",
  "nested_1.k2": "test",
  "nested_1.nested_2.k3": "56",
  "sample_rate": "0.7",
}
```

The adapted handling also ensures order on nested keys for calculating the hash sum